peoplescoaches/unix-shell
# issue 6 - add test, reorg files, and update .gitignore/makefile

What's changed:

- The test in `test_shell.c` has been updated to verify that the initial shell prompt will be displayed in the console upon running the executable (e.g. `pcshell> test` will produce "pcshell> test\n" as output)
- `main.c` was created to separate out the functionality of executing the shell into a separate file (`shell.c`). This allows the Unity testing framework to compile the tests using the `main()` function within `test_shell.c` without it conflicting with the `main()` function in `main.c`.
- The header file `shell.h` was created to provide functionality within `shell.c` to other files, i.e. `test_shell.c`
- The `build` directory was created to hold local files that are created in the compile and test processes
- Files have been organized into `src` and `test` directories
- The `makefile` has been updated with new file structure
- `.gitignore` now includes txt files that are created within `build` directory

To test:

- Pull down branch (and run `make clean` and `make compile` if needed)
- Run `make test` to execute the test
- Temporarily change the assertion on line 44 in `test_shell.c` to make the test fail
- Manually verify `pcshell` still works on your machine and that I didn't break it!

Future issue:
- The current implementation can only read the first line of output to the console. I will create another issue to create an additional test (or update this test) that captures all responses to the shell command. (e.g. `pcshell> test` should both produce "pcshell> test\n" and "test\n" as two lines of output)

Resolves #6
